aptos-move/block-executor/src/captured_reads.rs

```diff
@@ -661,7 +658,7 @@ where
         }
 
         self.module_reads.iter().all(|(key, read)| match read {
-            ModuleRead::GlobalCache => global_module_cache.contains_valid(key),
+            ModuleRead::GlobalCache(_) => global_module_cache.contains_valid(key),
```

Comment:
should this whole match be equivalent to:



```
        self.module_reads.iter().all(|(key, read)| {
            let previous_version = match read {
              ModuleRead::GlobalCache(_) => None, // i.e. storage version
              ModuleRead::PerBlockCache(previous) => previous.as_ref().map(|(_, version)| *version);
            };
            let current_version = per_block_module_cache.get_module_version(key);
            current_version == previous_version
        })
```

why do we need to update GlobalCache at all while executing a block?

Reply:
We do if we read first from it (to know if entry is overridden or not). An alternative is to check lower level cache first, but this means performance penalty due to locking.

Reply:
The code can be somewhat equivalent, but:
```
let current_version = per_block_module_cache.get_module_version(key);
```
causes a prefetch of storage version by default. We would need to special case validation to not do it. An we also end up locking the cache (shard, worst case), instead of checking an atomic bool

Reply:
this is because we may publish a module that invalidates the global cache that's being read I think